pkg/config/v2/parameter/compound/compound.go

```diff
@@ -0,0 +1,165 @@
+	Serializer:   writeCompoundParameter,
+	Deserializer: parseCompoundParameter,
+}
+
+type CompoundParameter struct {
+	Format               *template.Template
+	ReferencedParameters []parameter.ParameterReference
+}
+
+// this forces the compiler to check if CompoundParameter is of type Parameter
+var _ parameter.Parameter = (*CompoundParameter)(nil)
+
+func (p *CompoundParameter) GetType() string {
+	return CompoundParameterType
+}
+
+func (p *CompoundParameter) GetReferences() []parameter.ParameterReference {
+	return p.ReferencedParameters
+}
+
+func (p *CompoundParameter) ResolveValue(context parameter.ResolveContext) (interface{}, error) {
+	compoundData := make(map[string]interface{})
+
+	for _, param := range p.ReferencedParameters {
+		compoundData[param.Property] = context.ResolvedParameterValues[param.Property]
+	}
+
+	out := bytes.Buffer{}
+	err := p.Format.Execute(&out, compoundData)
+
+	if err != nil {
+		return nil, fmt.Errorf("error resolving compound value: %v", err)
+	}
+
+	return out.String(), nil
+
+}
+
+// parseCompoundParameter parses a given context into an instance of CompoundParameter.
+// This requires a string `format` and a slice of strings `references`, where `format`
+// is a template string and `references` are all the used references in `format` refering
+// to other parameters within the config.
+func parseCompoundParameter(context parameter.ParameterParserContext) (parameter.Parameter, error) {
+	formatInterface, ok := context.Value["format"]
+	if !ok {
+		return nil, createParameterParseError(context, "missing property `format`")
+	}
+
+	format, err := template.New(context.ParameterName).Option("missingkey=error").Parse(util.ToString(formatInterface))
+	if err != nil {
+		return nil, createParameterParseError(context, "format is not a valid template: %v", err)
+	}
+
+	references, ok := context.Value["references"]
+	if !ok {
+		return nil, createParameterParseError(context, "missing property `references`")
+	}
+
+	referencedParameterSlice, ok1 := references.([]interface{})
+	if !ok1 {
+		return nil, createParameterParseError(context, "malformed value `references`")
+	}
+	referencedParameters := toParameterReferences(referencedParameterSlice, context.Coordinate)
+
+	return &CompoundParameter{
+		Format:               format,
+		ReferencedParameters: referencedParameters,
+	}, nil
+}
+
+func writeCompoundParameter(context parameter.ParameterWriterContext) (map[string]interface{}, error) {
+	compoundParam, ok := context.Parameter.(*CompoundParameter)
+
+	if !ok {
+		return nil, &parameter.ParameterWriterError{
+			Location: context.Coordinate,
+			EnvironmentDetails: errors.EnvironmentDetails{
+				Group:       context.Group,
+				Environment: context.Environment,
+			},
+			ParameterName: context.ParameterName,
+			Reason:        "unexpected type. parameter is not of type `CompoundParameter`",
+		}
+	}
+
+	result := make(map[string]interface{})
+
+	format := util.ToString(compoundParam.Format)
```

Comment:
Will this always work? 
It might be beneficial to simply store the raw format string 
we get in the parse function in the parameter struct.

Reply:
Good point. As a matter of fact, the `ToString` would have never worked at all ^^